plugin/path_config.go

```diff
@@ -64,36 +102,28 @@ func (b *GcpAuthBackend) pathConfigWrite(ctx context.Context, req *logical.Reque
 	}
 
 	c, err := b.config(ctx, req.Storage)
-
 	if err != nil {
 		return nil, err
 	}
-	if c == nil {
```

Comment:
I stripped all the checks for whether the config is nil because I made it so the config is _always_ returned by the `b.config(ctx, req.Storage)` method. I did that because even if they haven't configured anything, they effectively _still_ have a config because the metadata fields are defaulting. This is consistent with the approach taken with AWS.

Reply:
Is this a safe assumption to make? I'm looking through to double check but I just want to be paranoid about this kind of change.

Reply:
Yes, I think it's good to be extra safe.

Just to be sure, I waited until I was fresh this morning and went through everywhere it's used again. I'm going to go through this PR and comment upon each location. 

For this location, it actually looks like [this would have panicked previously](https://github.com/hashicorp/vault-plugin-auth-gcp/blob/d0062d35cf3848e2e912a0d6f4afa214390b8bf4/plugin/backend.go#L174) if the config was nil because it's immediately used below. So, this code represents a behavior change. In looking at the code now, I think it actually fixes that bug and now does what's intended, which is to continue on to the default credentials below. So, this should be a safe change and actually have a positive effect.

Reply:
For this location, the effect of the config never being nil is basically no change since it was created below if it was nil.

Reply:
`formatAndMarshalCredentials` is nil safe: https://github.com/hashicorp/vault-plugin-auth-gcp/blob/master/plugin/path_config.go#L145

Reply:
Also, having the returned config be non-nil solves a bug [here](https://github.com/hashicorp/vault-plugin-auth-gcp/blob/d0062d35cf3848e2e912a0d6f4afa214390b8bf4/plugin/backend.go#L174) where it would have panicked if the config were nil. Now it will continue on to using the default credentials as the code intends.